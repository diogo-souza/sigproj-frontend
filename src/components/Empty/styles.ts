import styled, { css } from 'styled-components';

export const Container = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  text-align: center;

  img {
    object-fit: cover;
  }
`;

export const Title = styled.h2`
  ${({ theme }) => css`
    color: ${theme.colors.primary};
    font-size: ${theme.font.sizes.xxlarge};
  `}
`;

export const Description = styled.p`
  ${({ theme }) => css`
    font-weight: ${theme.font.light};
    color: ${theme.colors.titleBlack};
    font-size: ${theme.font.sizes.large};
    margin-bottom: ${theme.spacings.medium};
  `}
`;
