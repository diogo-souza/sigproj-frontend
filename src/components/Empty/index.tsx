import React from 'react';
import notFoundImg from 'assets/images/notFound.svg';
import { useHistory } from 'react-router-dom';
import Button from 'components/Button';
import * as S from './styles';

export type EmptyProps = {
  title: string;
  description: string;
  hasLink?: boolean;
  path?: string;
  text?: string;
};

const Empty: React.FC<EmptyProps> = ({
  title,
  description,
  hasLink = false,
  path = '/',
  text = 'Ir para inicio',
}: EmptyProps) => {
  const { push } = useHistory();
  return (
    <S.Container>
      <img
        src={notFoundImg}
        alt="Uma mulher nÃ£o encontrando um arquivo dentro de uma pasta"
        width={500}
      />
      <S.Title>{title}</S.Title>
      <S.Description>{description}</S.Description>

      {hasLink && <Button onClick={() => push(path)}>{text}</Button>}
    </S.Container>
  );
};
export default Empty;
