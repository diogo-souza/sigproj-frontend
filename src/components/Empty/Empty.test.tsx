import { fireEvent, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { renderWithTheme } from 'utils/tests/helpers';

import Empty from '.';

const mockHistoryPush = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useHistory: () => ({
    push: mockHistoryPush,
  }),
}));

const props = {
  title: 'A simple title',
  description: 'A simple description',
};

describe('<Empty />', () => {
  it('should render/navigating correctly', () => {
    const { container } = renderWithTheme(<Empty {...props} hasLink />);

    expect(
      screen.getByRole('img', {
        name: /Uma mulher nÃ£o encontrando um arquivo dentro de uma pasta/i,
      }),
    ).toBeInTheDocument();

    expect(
      screen.getByRole('heading', { name: /a simple title/i }),
    ).toBeInTheDocument();

    expect(screen.getByText(/a simple description/i)).toBeInTheDocument();

    const button = screen.getByRole('button', { name: /ir para inicio/i });
    expect(button).toBeInTheDocument();
    fireEvent.click(button);
    expect(mockHistoryPush).toHaveBeenCalledWith('/');

    expect(container.parentElement).toMatchSnapshot();
  });

  it('should not render link when hasLink is not passed', () => {
    renderWithTheme(<Empty {...props} />);

    expect(
      screen.queryByRole('button', { name: /ir para inicio/i }),
    ).not.toBeInTheDocument();
  });
});
