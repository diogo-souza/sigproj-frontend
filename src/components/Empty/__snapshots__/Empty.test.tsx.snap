// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Empty /> should render/navigating correctly 1`] = `
.c3 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: #B51222;
  color: #FFFFFF;
  border: 0;
  border-radius: 0.4rem;
  padding: 0.8rem;
  height: 4rem;
  font-size: 1.4rem;
  padding: 0.8rem 3.2rem;
  -webkit-transition: background-color 0.2s;
  transition: background-color 0.2s;
}

.c3:focus {
  box-shadow: 0 0 0 3px rgba(75,174,227,0.4);
  border-color: #0d6efd;
}

.c3:hover {
  background-color: #870d19;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  text-align: center;
}

.c0 img {
  object-fit: cover;
}

.c1 {
  color: #B51222;
  font-size: 2.8rem;
}

.c2 {
  font-weight: 300;
  color: #171923;
  font-size: 1.8rem;
  margin-bottom: 3.2rem;
}

<body>
  <div>
    <div
      class="c0"
    >
      <img
        alt="Uma mulher nÃ£o encontrando um arquivo dentro de uma pasta"
        src="notFound.svg"
        width="500"
      />
      <h2
        class="c1"
      >
        A simple title
      </h2>
      <p
        class="c2"
      >
        A simple description
      </p>
      <button
        class="c3"
        color="primary"
      >
        <span>
          Ir para inicio
        </span>
      </button>
    </div>
  </div>
</body>
`;
